src/common/types/row_data_collection.cpp

```diff
@@ -128,6 +134,11 @@ void RowDataCollection::Merge(RowDataCollection &other) {
 	for (auto &handle : temp.pinned_blocks) {
 		pinned_blocks.emplace_back(move(handle));
 	}
+	other.Clear();
+}
+
+unique_ptr<RowDataCollection> RowDataCollection::CloneEmpty(bool keep_pinned_p = false) {
+	return make_unique<RowDataCollection>(buffer_manager, block_capacity, entry_size, keep_pinned_p);
```

Comment:
I don't think this was me originally, but we should keep an eye on this because I'm not clear why it shouldn't just use `this->keep_pinned`.

Reply:
I have an in-memory block collection and a spillable block collection. The in-memory block collection needs to keep the blocks pinned because the pointers need to stay valid. The spillable block collection does not need to keep them pinned, because it will contain only aligned data/heap blocks with offsets, no pointers.